### PR TITLE
Fix missing RADIUS/CoA binding layer

### PR DESCRIPTION
**Pull request by jpereira**
As described in the rfc5176, the CoA uses the port 3799

### CODE REVIEW

**Comment by jpereira** [comment]
https://tools.ietf.org/html/rfc5176#section-2.3

```
2.3.  Packet Format

   For either Disconnect-Request or CoA-Request packets UDP port 3799 is
   used as the destination port.  For responses, the source and
   destination ports are reversed.  Exactly one RADIUS packet is
   encapsulated in the UDP Data field.
```

File: scapy/layers/radius.py [hunk 1]
```diff
@@ -1154,3 +1154,5 @@ def post_build(self, p, pay):
 bind_layers(UDP, Radius, dport=1812)
 bind_layers(UDP, Radius, sport=1813)
 bind_layers(UDP, Radius, dport=1813)
+bind_layers(UDP, Radius, sport=3799)
+bind_layers(UDP, Radius, dport=3799)
```
Comment (gpotter2) [inline scapy/layers/radius.py]: Could you rather use
```
bind_bottom_up(UDP, Radius, sport=3799)
bind_bottom_up(UDP, Radius, dport=3799)
bind_layers(UDP, Radius, sport=3799, dport=3799)
```
And update the ones above ?

You only need a single `bind_layers` in total, which will be the default when building the packets. All others are only used for dissection

Thanks

Comment (jpereira) [reply scapy/layers/radius.py]: Ah! Got it, thank you @gpotter2 . I updated the pr.

Comment (jpereira) [reply scapy/layers/radius.py]: @gpotter2 so, I did some tests as you suggested but didn't work.

e.g:

1. Not work.

```python
bind_bottom_up(UDP, Radius, sport=1812)
bind_bottom_up(UDP, Radius, dport=1812)
bind_bottom_up(UDP, Radius, sport=1813)
bind_bottom_up(UDP, Radius, dport=1813)
bind_bottom_up(UDP, Radius, sport=3799)
bind_bottom_up(UDP, Radius, dport=3799)
bind_layers(UDP, Radius, sport=3799, dport=3799)
```

But, that one works pretty well.

```python
bind_layers(UDP, Radius, sport=1812)
bind_layers(UDP, Radius, dport=1812)
bind_layers(UDP, Radius, sport=1813)
bind_layers(UDP, Radius, dport=1813)
bind_layers(UDP, Radius, sport=3799)
bind_layers(UDP, Radius, dport=3799)
```

btw, I noticed other layers using multiple `bind_layers()` for different ports as the `dns.py`. therefore, what do you suggest?

Comment (jpereira) [reply scapy/layers/radius.py]: Ah! Got it, thank you @gpotter2 . I updated the pr.

**Comment by gpotter2** [comment]
Thanks.
There's a test failing because it used the previous default value. You can either change the test or keep the previous default value

**Comment by jpereira** [comment]
@gpotter2 Better to keep the default as 1812. I update it. thanks

File: scapy/layers/radius.py [hunk 2]
```diff
@@ -1150,7 +1150,10 @@ def post_build(self, p, pay):
...
-bind_layers(UDP, Radius, dport=1812)
-bind_layers(UDP, Radius, sport=1813)
-bind_layers(UDP, Radius, dport=1813)
+bind_bottom_up(UDP, Radius, sport=1812)
+bind_bottom_up(UDP, Radius, dport=1812)
+bind_bottom_up(UDP, Radius, sport=1813)
+bind_bottom_up(UDP, Radius, dport=1813)
+bind_bottom_up(UDP, Radius, sport=3799)
+bind_bottom_up(UDP, Radius, dport=3799)
+bind_layers(UDP, Radius, sport=1812, dport=1812)
```
Comment (gpotter2) [inline scapy/layers/radius.py]:
```suggestion
bind_layers(UDP, Radius, sport=1813, dport=1813)
```

Comment (jpereira) [reply scapy/layers/radius.py]: The major port is 1812 used to authentication, 1813 is for accounting and 3799 for CoA. But, ok. I updated the branch.

Comment (gpotter2) [reply scapy/layers/radius.py]: Feel free to do whatever is best. Just remember to update the test afterwards

Comment (jpereira) [reply scapy/layers/radius.py]: But, now is green. right?

**Comment by gpotter2** [comment]
I'm sorry I wasn't clear. As you can see, tests fail with:

```
Regression tests for Scapy
-- Run Thu Nov 28 18:16:17 2019 from [test\regression.uts] by UTscapy

Passed=1369
Failed=1


######
### RADIUS tests
######


###(1225)=[failed] IP/UDP/RADIUS - Build

>>> s = raw(IP()/UDP(sport=1812)/Radius(authenticator="scapy")/RadiusAttribute(value="scapy"))
>>> s == b'E\x00\x007\x00\x01\x00\x00@\x11|\xb3\x7f\x00\x00\x01\x7f\x00\x00\x01\x07\x14\x07\x15\x00#U\xb2\x01\x00\x00\x1bscapy\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x07scapy'
False
```

You'll have to update the following test:
https://github.com/secdev/scapy/blob/master/test/regression.uts#L10583-L10585
With whatever values are now required

**Comment by jpereira** [comment]
Got it. I updated the test and looks good now.

**Pull request merged**